setup.py

```diff
@@ -1,187 +1,71 @@
-"""
-Much of the build system code was adapted from work done by the pandas
-developers [1], which was in turn based on work done in pyzmq [2] and lxml [3].
-
-[1] http://pandas.pydata.org
-[2] http://zeromq.github.io/pyzmq/
-[3] http://lxml.de/
-"""
-
+import fnmatch
 import os
+from collections import defaultdict
 from os.path import relpath, join as pjoin
-import sys
-import subprocess
-import re
-from distutils.version import StrictVersion, LooseVersion
-
 
-# temporarily redirect config directory to prevent matplotlib importing
-# testing that for writeable directory which results in sandbox error in
-# certain easy_install versions
-os.environ["MPLCONFIGDIR"] = "."
+import pkg_resources
 
-no_frills = (len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
-                                     sys.argv[1] in ('--help-commands',
-                                                     'egg_info', '--version',
-                                                     'clean')))
-
-# try bootstrapping setuptools if it doesn't exist
 try:
-    import pkg_resources
-    try:
-        pkg_resources.require("setuptools>=0.6c5")
-    except pkg_resources.VersionConflict:
-        from ez_setup import use_setuptools
-        use_setuptools(version="0.6c5")
-    from setuptools import setup, Command, find_packages
-    _have_setuptools = True
+    # SM_FORCE_C is a testing shim to force setup to use C source files
+    FORCE_C = int(os.environ.get('SM_FORCE_C', 0))
+    if FORCE_C:
+        raise ImportError('Force import error for testing')
+    from Cython import Tempita
+    from Cython.Build import cythonize
+    from Cython.Distutils import build_ext
+
+    HAS_CYTHON = True
 except ImportError:
-    # no setuptools installed
-    from distutils.core import setup, Command
-    _have_setuptools = False
+    from setuptools.command.build_ext import build_ext
```

Comment:
Is this still for the "no setuptools installed" case?  Is that still possible in this day and age?

Reply:
There are only 2 cases, Cython installed or setuptools installed.

Reply:
Yep, reading comprehension fail on my part, thanks.